tests/suit/openstack_test.go

```diff
@@ -19,6 +19,63 @@ const (
 var _ = Describe("[level:component]Migration tests for OpenStack provider", func() {
 	f := framework.NewFramework("migration-func-test")
 
+	It("[test] should create OSP provider using https/ssl with CA", func() {
+		namespace := f.Namespace.Name
+		err := f.Clients.OpenStackClient.SetupClient("cirros-volume", "net-int", "nfs")
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Load Source VM Details from OpenStack")
+		packstackCA, err := f.Clients.OpenStackClient.LoadCA(f, packstackNameSpace, "packstack")
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create Secret from Definition")
+		s, err := utils.CreateSecretFromDefinition(f.K8sClient, utils.NewSecretDefinition(nil,
+			map[string]string{
+				"username":    "admin",
+				"password":    "12e2f14739194a6c",
+				"domainName":  "default",
+				"projectName": "admin",
+				"regionName":  "RegionOne",
+				"cacert":      packstackCA,
+			}, nil, namespace, "os-test-secret"))
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create osp provider")
+		pr := utils.NewProvider(openstackProviderName+"-ssl-secure", forkliftv1.OpenStack, namespace, map[string]string{},
+			"https://packstack.konveyor-forklift:5001/v3", s)
+		err = utils.CreateProviderFromDefinition(f.CrClient, pr)
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitForProviderReadyWithTimeout(f.CrClient, namespace, openstackProviderName+"-ssl-secure", 30*time.Second)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("[test] should create OSP provider using https/ssl insecure", func() {
+		namespace := f.Namespace.Name
+		err := f.Clients.OpenStackClient.SetupClient("cirros-volume", "net-int", "nfs")
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create Secret from Definition")
+		s, err := utils.CreateSecretFromDefinition(f.K8sClient, utils.NewSecretDefinition(nil,
+			map[string]string{
+				"username":           "admin",
+				"password":           "12e2f14739194a6c",
+				"domainName":         "default",
+				"projectName":        "admin",
+				"regionName":         "RegionOne",
+				"insecureSkipVerify": "true",
+				"cacert":             "",
```

Comment:
I try to see if I got it right - the difference between this and the provider that is set below is in this line, did you add this test to check adding a provider separately (and not as part of testing it with NetworkMap) or because of this line?

Reply:
this only tests provider creation, with out running the migration, i can simplify this further and only test the openstack.client  interface  (without creating the provider CRs ie) so the  connection attempt will be made from the test  code (outside from the cluster) and not from the inventory pod.

Reply:
the name `[test] should create provider with NetworkMap` below is misleading and should be changed.

Reply:
done